src/browser-frontend/views/browser/window.jsx

```diff
@@ -10,23 +10,49 @@ CONDITIONS OF ANY KIND, either express or implied. See the License for the
 specific language governing permissions and limitations under the License.
 */
 
-import React, { PureComponent } from 'react';
+import React, { PureComponent, PropTypes } from 'react';
 import CSSModules from 'react-css-modules';
+import { connect } from 'react-redux';
+import Mousetrap from 'react-mousetrap-mixin';
+
+import PagesModelActions from '../../actions/pages-model-actions';
+
 
 import Styles from './window.css';
 import Chrome from './chrome';
 import Content from './content';
 
+@connect(() => ({
+}))
 @CSSModules(Styles, {
   allowMultiple: true,
 })
 export default class Window extends PureComponent {
   render() {
+    const keys = {};
+    const mousetrap = (<Mousetrap mousetrap={keys} />);
+
+    mousetrap.type.prototype.addMousetrap(['command+t', 'ctrl+t'], () => {
+      this.props.dispatch(PagesModelActions.addPage());
+    });
+    mousetrap.type.prototype.addMousetrap(['command+w', 'ctrl+w'], () => {
+      this.props.dispatch(PagesModelActions.removePage({}));
+    });
+    mousetrap.type.prototype.addMousetrap('up up down down left right left right b a', () => {
+      const url = 'http://chilloutandwatchsomecatgifs.com/';
+      this.props.dispatch(PagesModelActions.addPage({ url }));
+    });
```

Comment:
Creating functions inside render methods which end up as props is discouraged in React, because it causes rerenders on the respective children, regardless of wether or not `shouldComponentUpdate` is used. This is specifically disallowed with some linting rules, but obviously these rules aren't so elaborate as to catch everything.

Now, I'm not sure what's happening exactly with `mousetrap`, and whether or not this actually affects performance in any way, but this looks like an antipattern, while it's seemingly easily avoidable to do any kind of work in this render function by just following the example here: https://www.npmjs.com/package/react-mousetrap-mixin#usage-example. The `mousetrap` object there could be created in the constructor.

That being said, is there any particular advantage to using `react-mousetrap-mixin` (a seemingly unpopular library) over `moustrap` directly?